test/smoke/run-smoke-suite.sh

```diff
@@ -120,26 +80,51 @@ function extra_debug_logs {
 
 }
 
-declare -a mandatory
-mandatory=(
-  CLIENT_ID
-  USER_TOKEN
-  SMOKE_REGISTRY_TOKEN
-  SMOKE_REGISTRY_SECRET
-)
-
-missing=false
-for var in "${mandatory[@]}"; do
-  if [[ -z "${!var:-}" ]]; then
-    echo "Environment variable $var must be set"
-    missing=true
-  fi
-done
-
-if [ "$missing" = true ]; then
-  exit 1
+if [[ -n "${DEBUG}" ]]; then
+  set -x
 fi
 
+set -o errexit
```

Comment:
What is the purpose of moving lines 3-41 to here?

Reply:
nad git history cleared )))

Reply:
or can we move all functions to `functions.sh` as we have this file now?

Reply:
moved